pkg/build/controller/image_change_controller_test.go

```diff
@@ -296,67 +287,51 @@ func TestNoImageIDChange(t *testing.T) {
 	buildcfg.Triggers[0].ImageChange.LastTriggeredImageID = "imageID123"
 	imagerepo := mockImageRepo("testImageRepo", "registry.com/namespace/imagename", map[string]string{"testTag": "imageID123"})
 	controller := mockImageChangeController(buildcfg)
-	err := controller.HandleImageRepo(imagerepo)
-	buildCreator := controller.BuildCreator.(*mockBuildCreator)
+	buildConfigInstantiator := controller.BuildConfigInstantiator.(*mockBuildConfigInstantiator)
 	buildConfigUpdater := controller.BuildConfigUpdater.(*mockBuildConfigUpdater)
 
+	err := controller.HandleImageRepo(imagerepo)
 	if err != nil {
 		t.Errorf("Unexpected error %v from HandleImageRepo", err)
 	}
-	if buildCreator.build != nil {
-		t.Error("New build created when no change happened!")
+	if len(buildConfigInstantiator.name) != 0 {
+		t.Error("New build generated when no change happened!")
 	}
 	if buildConfigUpdater.buildcfg != nil {
 		t.Error("BuildConfig was updated when no change happened!")
 	}
 }
 
-func TestBuildCreateError(t *testing.T) {
+func TestBuildConfigInstantiatorError(t *testing.T) {
 	// valid configuration, but build creation fails, in that situation the buildconfig should not be updated
 	buildcfg := mockBuildConfig("registry.com/namespace/imagename", "registry.com/namespace/imagename", "testImageRepo", "testTag")
 	imagerepo := mockImageRepo("testImageRepo", "registry.com/namespace/imagename", map[string]string{"testTag": "newImageID123"})
 	controller := mockImageChangeController(buildcfg)
-	buildCreator := controller.BuildCreator.(*mockBuildCreator)
-	buildCreator.err = fmt.Errorf("error")
-	err := controller.HandleImageRepo(imagerepo)
-	buildConfigUpdater := controller.BuildConfigUpdater.(*mockBuildConfigUpdater)
+	buildConfigInstantiator := controller.BuildConfigInstantiator.(*mockBuildConfigInstantiator)
+	buildConfigInstantiator.err = fmt.Errorf("instantiating error")
 
-	if err == nil {
+	err := controller.HandleImageRepo(imagerepo)
+	if err == nil || !strings.Contains(err.Error(), "instantiating error") {
 		t.Error("Expected error from HandleImageRepo")
 	}
-	if _, ok := err.(ImageChangeControllerFatalError); ok {
-		t.Error("Expected retryable error from HandleImageRepo")
-	}
-	if buildCreator.build == nil {
-		t.Fatal("Expected new build when new image was created!")
-	}
-	if buildCreator.build.Parameters.Strategy.DockerStrategy.Image != "registry.com/namespace/imagename:newImageID123" {
-		t.Errorf("Image substitutions not properly setup for new build.  Expected %s, got %s |", "registry.com/namespace/imagename:newImageID123", buildCreator.build.Parameters.Strategy.DockerStrategy.Image)
-	}
-	if buildConfigUpdater.buildcfg != nil {
-		t.Fatal("Expected no buildConfig update on BuildCreate error!")
```

Comment:
again it seems like you've lost part of this test...  we wannt to confirm that BuildConfig is not being updated when we get an error on build create.